internal/operator-controller/contentmanager/cache/cache.go

```
@@ -233,9 +238,10 @@ func (c *cache) getCacheGVKs() sets.Set[schema.GroupVersionKind] {
 //
 // An empty Group is assumed to be the "core" Kubernetes
 // API group.
-func gvksForObjects(objs ...client.Object) (sets.Set[schema.GroupVersionKind], error) {
-	gvkSet := sets.New[schema.GroupVersionKind]()
+func sourceKeysForObjects(objs ...client.Object) (sets.Set[sourceKey], error) {
+	sourceKeys := sets.New[sourceKey]()
 	for _, obj := range objs {
+		ns := obj.GetNamespace()
```

I would recommend gating this in a check for the scope of the resource (namespace vs cluster). Don't set a namespace for cluster scoped resource informers.

For example, if I were to provide you a `ClusterRole` defined like:
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: role-grantor
  namespace: default
rules:
- apiGroups: ["rbac.authorization.k8s.io"]
  resources: ["rolebindings"]
  verbs: ["create"]
- apiGroups: ["rbac.authorization.k8s.io"]
  resources: ["clusterroles"]
  verbs: ["bind"]
  # omit resourceNames to allow binding any ClusterRole
  resourceNames: ["admin","edit","view"]
```

The informer you create here would not be valid for monitoring clusterroles as they are cluster scoped resources. Even though specifying a namespace here does nothing, it is accepted by the Kube apiserver.

If you get these objects from parsing manifests that may result in you having a namespace set for a cluster scoped resource I would perform this check. If you are only ever going to get these objects from the cluster, it _might_ be OK to just statically set this, but it doesn't hurt to have the extra layer of protection anyways.

---

Good catch! This input _could_ be coming directly from user-provided manifests, so best to sanitize it. For now, I coded it to silently ignore the namespace setting in the manifest for cluster-scoped objects.

It is maybe debatable if we should consider that an error. I guess since the API server allows it, we should too?